drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c

```diff
@@ -433,21 +433,27 @@ int bcmf_driver_download_clm(FAR struct bcmf_dev_s *priv)
 #endif
 #endif /* CONFIG_IEEE80211_BROADCOM_HAVE_CLM */
 
-int bcmf_driver_initialize(FAR struct bcmf_dev_s *priv)
+int bcmf_wl_active(FAR struct bcmf_dev_s *priv, bool active)
 {
-  int ret;
+  int interface = CHIP_STA_INTERFACE;
+  uint8_t tmp_buf[64];
   uint32_t out_len;
   uint32_t value;
-  uint8_t tmp_buf[64];
-  int interface = CHIP_STA_INTERFACE;
+  int ret;
+
+  ret = bcmf_bus_sdio_active(priv, active);
+  if (ret != OK || !active)
+    {
+      return ret;
+    }
 
 #ifdef CONFIG_IEEE80211_BROADCOM_HAVE_CLM
   /* Download CLM blob if needed */
 
   ret = bcmf_driver_download_clm(priv);
   if (ret != OK)
     {
-      return -EIO;
+      return ret;
```

Comment:
should we call bcmf_bus_sdio_active(false) in case of fail?

Reply:
Done

Reply:
how about other fail case?

Reply:
Done